projects/client/RabbitMQ.Client/src/client/impl/Connection.cs

```diff
@@ -107,6 +108,16 @@ public class Connection : IConnection
         private AutoResetEvent m_heartbeatRead = new AutoResetEvent(false);
         private AutoResetEvent m_heartbeatWrite = new AutoResetEvent(false);
 
+#if CORECLR
+        private static string version = typeof(Connection).GetTypeInfo().Assembly
```

Comment:
Why in the world would they change what `typeof()` returns?!?! [sigh]

Reply:
In .NET Core 1.0, there was an attempt to split out reflecton-oriented APIs, and `GetTypeInfo` was part of that.

In ,NET Core 2.0 / .NET Standard 2.0, they've fixed all of that, so you wouldn't need to do that if you moved up to `netstandard2.0` at some point.

Reply:
Thanks for the explanation... I didn't expect to see a response.

@kjnilsson maybe we should move on up to 2.0 then? FWIW, that's the version that I have on my workstation (Arch Linux)

Reply:
[.NET Standard versions](https://github.com/dotnet/standard/blob/master/docs/versions.md) are a trade off. Lower versions = wider platform support but a smaller API surface. Higher versions = narrower platform support but a larger API surface.

By moving from .NET Standard 1.5 to 2.0, you'll be increasing the required versions for a few platforms:

- .NET Core: 1.0 → 2.0
- Mono: 4.6 → 5.4
- Xamarin.iOS: 10.0 → 10.14
- Xamarin.Mac: 3.0 → 3.8
- Xamarin.Android: 7.0 → 8.0

The most significant of those is clearly .NET Core 1.0 → 2.0. I guess the others are not significant.